### PR TITLE
fix(form): adapt the form with the guideline

### PR DESCRIPTION
**Pull request by jare-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
The font-size of one input or one label is smaller.
The style for the disabled/readonly input is incorrect
No highlight on the label when the user focused one input
When the input is filled, the label is lighter
When the input has a placeholder, the title is removed.

Old form:
initial_form

**What is the new behavior?**

form_focus
forms

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

File: packages/theme/src/theme/_forms.scss [hunk 1]
```diff
@@ -307,30 +327,47 @@ form {
...
+			font-size: $font-size-small;
 		}
 	}
 
-	&-control[placeholder]:not([placeholder='']) {
+	&-control:focus,
+	{
 		+ label {
-			opacity: 0;
+			opacity: 1 !important;
```
Comment (talend-glorieux) [inline packages/theme/src/theme/_forms.scss]: Please remove the important.

Comment (jare-talend) [inline packages/theme/src/theme/_forms.scss]: done. I'm going to see that for the element textarea, select we can't detect if the user has set a value. I'm going to see with @maximenicoul if the state is relevant.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

File: output/theme.sasslint.txt [hunk 2]
```diff
@@ -12,21 +12,30 @@ src/theme/_dropdowns.scss
...
+  275:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+  288:8   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+  311:9   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
+  337:20  warning  Whitespace required before {                             space-before-brace
+  340:1   warning  Space expected between blocks                            empty-line-between-blocks
+  340:20  warning  Whitespace required before {                             space-before-brace
+  353:29  warning  Whitespace required before {                             space-before-brace
+  360:19  warning  Whitespace required before {                             space-before-brace
+  361:9   warning  Whitespace required before {                             space-before-brace
+  414:4   warning  Vendor prefixes should not be used                       no-vendor-prefixes
```
Comment (frassinier) [inline output/theme.sasslint.txt]: Please do not introduce more regressions

Comment (jare-talend) [inline output/theme.sasslint.txt]: I see that. It made on the last commit. I'm waiting the next travis

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://275.talend.surge.sh)

**Pull request merged**